### ISSUE TITLE
Problem enforcing RAM limit

### ISSUE THREAD

**Issue by jernsting** [automl/auto-sklearn#720]
I am using the latest version of autosklearn and running it on a high performance cluster.

Here is my code:
```
automl = autosklearn.classification.AutoSklearnClassifier(ensemble_size=1,
               initial_configurations_via_metalearning=0,
               time_left_for_this_task=14400,
               tmp_folder='[removed]'),
               ml_memory_limit=65536,
               n_jobs=4,
               resampling_strategy='holdout',
               resampling_strategy_arguments={'train_size': 0.5},
               output_folder='[removed]'),
               disable_evaluator_output=False)
```

The ml_memory_limit is set to 64 GB but the program crashes with 70 GB RAM assigned by the hpc. Shouldn't be 65 GB enough to succeed for the run?

**Comment by mfeurer** [comment]
The memory limit is only for the target algorithm, i.e. the machine learning pipeline being trained. Auto-sklearn itself and the Bayesian optimization toolkit SMAC also require some memory, although, this cannot be limited. To be on the safe side, you can budget a few gigabytes in (usually, 2-4 are enough) to make sure the HPC scheduler doesn't kill your jobs.
